src/Stratis.Bitcoin.Features.PoA/FederationManager.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Configuration;
+
+namespace Stratis.Bitcoin.Features.PoA
+{
+    public class FederationManager
+    {
+        /// <summary><c>true</c> in case current node is a federation member.</summary>
+        public bool IsFederationMember { get; private set; }
+
+        /// <summary>Key of current federation member. <c>null</c> if <see cref="IsFederationMember"/> is <c>false</c>.</summary>
+        public Key FederationMemberKey { get; private set; }
+
+        private readonly NodeSettings settings;
+
+        private readonly PoANetwork network;
+
+        private readonly ILogger logger;
+
+        public FederationManager(NodeSettings nodeSettings, Network network, ILoggerFactory loggerFactory)
+        {
+            this.settings = nodeSettings;
+            this.network = network as PoANetwork;
+
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+        }
+
+        public void Initialize()
+        {
+            this.LoadKey();
+
+            if (this.FederationMemberKey != null)
```

Comment:
If you are a federation member how can this be null?

Reply:
When the node loads it should decide if we are federation member or not. It does that by loading a key. if the key is not there- we are normal node. 
So here we tried to load the key and if it's null we set FederationMemberKey to null and IsFederationMember to false. Otherwise we set 2 values. 

Later miner will check if we are fed member and start mining if we are. 

that way we don't need much changes in the code or user interaction to become a fed member.